deploy/deploy.go

```diff
@@ -269,58 +288,51 @@ func doSingleUpload(ctx context.Context, bucket *blob.Bucket, upload *fileToUplo
 // localFile represents a local file from the source. Use newLocalFile to
 // construct one.
 type localFile struct {
-	// Path is the relative path to the file.
-	Path string
+	// NativePath is the native path to the file (using file.Separator).
+	NativePath string
+	// SlashPath is NativePath converted to use /.
+	SlashPath string
 	// UploadSize is the size of the content to be uploaded. It may not
 	// be the same as the local file size if the content will be
 	// gzipped before upload.
 	UploadSize int64
-	// UploadContentReader reads the content to be uploaded. Again,
-	// it may not be the same as the local file content due to gzipping.
-	UploadContentReader io.Reader
```

Comment:
UploadContentReader did not play nice with Windows, because it created an open reader based on a file that was never closed (it was opened in the old `contentToUpload`, and can't be closed there without making the `Reader` not work anymore). Windows is persnickity about deleting files that have been opened.

The new approach is to read the file contents into a `byte.Buffer` regardless, and create a `Reader` as needed. This also cleans up the code a bit when computing MD5 hashes, but it does mean we'll read everything into memory. I can optimize this by trying to do the read lazily if you think it's worth it.

Reply:
Why not set `UploadContentReader`to a `hugio.ReadSeekCloser`? And 1) reset the reader (seek) every time you use it and 2) Close every `localFile` at the end.

There can be lots of potentially big files here and we should not need to have 20 GB of ram to publish 10 GB of files. There may be double work (that may be optimized away) with the above approach, but we should try to avoid Out of mem errors at all cost.

Reply:
I changed `Reader()` to return an `io.ReadCloser` that the caller must close. Closing a returned reader is easier to manage than trying to close all the `localFile` objects.